src/backend/utils/cache/inval.c

```diff
@@ -1063,6 +1087,7 @@ AtEOXact_Inval(bool isCommit)
 
 	/* Need not free anything explicitly */
 	transInvalInfo = NULL;
+	ClearSavedCatcacheMessages();
```

Comment:
can we do this via hook? Just to be extra safe so that in case we have some issue with savedCatcacheMessages memory allocation, it does not leak outside babelfish

Reply:
this is not straightforward. because queryEnvironment is defined in `queryEnvironment.c`, we can't really reference internal members of the struct outside `queryenvironment.c`. it can work if we expose it to the header file, but that also means altering PG code.